Error on fish_prompt function when history doesn't exist

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

when [fish_prompt](https://github.com/JanDeDobbeleer/oh-my-posh/blob/393f30a082e96f7341b1e4213e80867bedeacab2/src/shell/scripts/omp.fish#L14) is invoked and `~/.local/share/fish/fish_history` doesn't exist the following error happen:

```
test: Missing argument at index 3
= clear
        ^
- (line 41): 
    if test (history | head -1) = "clear"
       ^
in function 'fish_prompt'
in command substitution
```

I got this error when running a docker container for the first time: a situation where history has not been created.

# Expected result:

The prompt to be initialized without any error message.

### Theme

pure.omp.json

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

fish

### Log output

```Shell
Version: 14.19.1

Segments:

ConsoleTitle(true)                         -   2 ms -  ~
session(true)                              -   0 ms - lompask 
path(true)                                 -   0 ms - ~ 
git(false)                                 -   0 ms - 
executiontime(false)                       -   0 ms - 
exit(true)                                 -   0 ms - ❯ 

Run duration: 6.193975ms

Cache path: /home/lompask

Config path: /home/lompask/.poshthemes/pure.omp.json

Logs:

[DEBUG] 11:13:25.774 shell.go:Getenv:392 → NO DATA
[TRACE] 11:13:25.774 shell.go:Getenv(TERM_PROGRAM) - 41.775µs
[DEBUG] 11:13:25.774 shell.go:Getenv:392 → NO DATA
[TRACE] 11:13:25.774 shell.go:Getenv(XDG_CACHE_HOME) - 6.572µs
[TRACE] 11:13:25.774 shell_unix.go:CachePath() - 32.346µs
[DEBUG] 11:13:25.774 shell.go:Getenv:392 → /home/lompask/.poshthemes/pure.omp.json
[TRACE] 11:13:25.774 shell.go:Getenv(POSH_THEME) - 10.014µs
[DEBUG] 11:13:25.774 shell_unix.go:Platform:63 → ubuntu
[TRACE] 11:13:25.774 shell.go:resolveConfigPath() - 24.751µs
[TRACE] 11:13:25.774 shell.go:Init() - 362.606µs
[TRACE] 11:13:25.774 shell.go:Flags() - 168ns
[TRACE] 11:13:25.775 config.go:loadConfig() - 1.090142ms
[TRACE] 11:13:25.775 shell.go:Flags() - 235ns
[DEBUG] 11:13:25.775 shell.go:Getenv:392 → NO DATA
[TRACE] 11:13:25.775 shell.go:Getenv(OMP_CACHE_DISABLED) - 10.911µs
[TRACE] 11:13:25.776 shell_unix.go:Root() - 2.555µs
[DEBUG] 11:13:25.777 shell.go:Shell:641 → process name: fish
[TRACE] 11:13:25.777 shell.go:Shell() - 1.28413ms
[TRACE] 11:13:25.777 shell.go:ErrorCode() - 165ns
[DEBUG] 11:13:25.777 shell.go:Getenv:392 → NO DATA
[TRACE] 11:13:25.777 shell.go:Getenv(WSL_DISTRO_NAME) - 5.21µs
[TRACE] 11:13:25.777 shell_unix.go:IsWsl() - 18.789µs
[TRACE] 11:13:25.777 shell.go:GOOS() - 169ns
[DEBUG] 11:13:25.777 shell.go:Pwd:420 → NO DATA
[TRACE] 11:13:25.777 shell.go:Pwd() - 30.668µs
[TRACE] 11:13:25.777 shell.go:GOOS() - 109ns
[TRACE] 11:13:25.777 shell.go:GOOS() - 213ns
[TRACE] 11:13:25.777 shell.go:GOOS() - 67ns
[TRACE] 11:13:25.777 shell.go:GOOS() - 69ns
[TRACE] 11:13:25.777 shell.go:GOOS() - 65ns
[TRACE] 11:13:25.777 shell.go:GOOS() - 67ns
[TRACE] 11:13:25.777 shell.go:GOOS() - 68ns
[TRACE] 11:13:25.777 shell.go:GOOS() - 67ns
[TRACE] 11:13:25.777 shell.go:GOOS() - 113ns
[DEBUG] 11:13:25.777 shell.go:User:545 → lompask
[TRACE] 11:13:25.777 shell.go:User() - 12.563µs
[DEBUG] 11:13:25.777 shell.go:Host:557 → 6e7afa9bed36
[TRACE] 11:13:25.777 shell.go:Host() - 9.223µs
[TRACE] 11:13:25.777 shell.go:GOOS() - 111ns
[DEBUG] 11:13:25.777 shell_unix.go:Platform:63 → ubuntu
[TRACE] 11:13:25.777 shell.go:TemplateCache() - 1.510298ms
[TRACE] 11:13:25.777 shell.go:Flags() - 152ns
[DEBUG] 11:13:25.777 shell.go:Pwd:400 → /home/lompask
[TRACE] 11:13:25.777 shell.go:Pwd() - 6.69µs
[TRACE] 11:13:25.777 shell.go:GOOS() - 113ns
[DEBUG] 11:13:25.777 shell.go:Getenv:392 → NO DATA
[TRACE] 11:13:25.777 shell.go:Getenv(SSH_CONNECTION) - 5.021µs
[DEBUG] 11:13:25.777 shell.go:Getenv:392 → NO DATA
[TRACE] 11:13:25.777 shell.go:Getenv(SSH_CLIENT) - 4.055µs
[TRACE] 11:13:25.777 shell.go:TemplateCache() - 180ns
[TRACE] 11:13:25.778 shell.go:TemplateCache() - 364ns
[TRACE] 11:13:25.778 shell.go:Shell() - 209ns
[DEBUG] 11:13:25.778 shell.go:Pwd:400 → /home/lompask
[TRACE] 11:13:25.778 shell.go:Pwd() - 12.098µs
[TRACE] 11:13:25.778 shell.go:GOOS() - 113ns
[DEBUG] 11:13:25.778 shell.go:Pwd:400 → /home/lompask
[TRACE] 11:13:25.778 shell.go:Pwd() - 5.509µs
[TRACE] 11:13:25.778 shell.go:Shell() - 88ns
[TRACE] 11:13:25.778 shell.go:Shell() - 115ns
[TRACE] 11:13:25.778 shell.go:GOOS() - 117ns
[TRACE] 11:13:25.778 shell.go:PathSeparator() - 127ns
[TRACE] 11:13:25.778 shell.go:GOOS() - 71ns
[TRACE] 11:13:25.778 shell.go:GOOS() - 65ns
[TRACE] 11:13:25.778 shell.go:GOOS() - 117ns
[TRACE] 11:13:25.778 shell.go:GOOS() - 74ns
[TRACE] 11:13:25.778 shell.go:PathSeparator() - 67ns
[TRACE] 11:13:25.778 shell.go:PathSeparator() - 68ns
[TRACE] 11:13:25.778 shell.go:GOOS() - 69ns
[TRACE] 11:13:25.778 shell.go:GOOS() - 67ns
[TRACE] 11:13:25.778 shell.go:PathSeparator() - 68ns
[TRACE] 11:13:25.778 shell.go:GOOS() - 66ns
[TRACE] 11:13:25.778 shell.go:GOOS() - 112ns
[TRACE] 11:13:25.778 shell.go:PathSeparator() - 83ns
[TRACE] 11:13:25.778 shell.go:GOOS() - 69ns
[TRACE] 11:13:25.778 shell.go:GOOS() - 67ns
[TRACE] 11:13:25.778 shell.go:PathSeparator() - 113ns
[DEBUG] 11:13:25.778 shell.go:Pwd:400 → /home/lompask
[TRACE] 11:13:25.778 shell.go:Pwd() - 5.022µs
[DEBUG] 11:13:25.778 shell.go:Getenv:392 → NO DATA
[TRACE] 11:13:25.778 shell.go:Getenv(WSL_DISTRO_NAME) - 4.678µs
[TRACE] 11:13:25.778 shell_unix.go:IsWsl() - 18.906µs
[TRACE] 11:13:25.778 shell.go:StackCount() - 143ns
[TRACE] 11:13:25.778 shell_unix.go:DirIsWritable(/home/lompask) - 14.824µs
[TRACE] 11:13:25.778 shell.go:TemplateCache() - 159ns
[TRACE] 11:13:25.779 shell.go:TemplateCache() - 354ns
[TRACE] 11:13:25.779 shell.go:Shell() - 179ns
[TRACE] 11:13:25.779 shell.go:Flags() - 158ns
[DEBUG] 11:13:25.779 shell.go:Pwd:400 → /home/lompask
[TRACE] 11:13:25.779 shell.go:Pwd() - 6.294µs
[TRACE] 11:13:25.779 shell.go:GOOS() - 112ns
[DEBUG] 11:13:25.779 shell.go:Getenv:392 → NO DATA
[TRACE] 11:13:25.779 shell.go:Getenv(WSL_DISTRO_NAME) - 4.281µs
[TRACE] 11:13:25.779 shell_unix.go:IsWsl() - 9.072µs
[TRACE] 11:13:25.779 shell_unix.go:IsWsl2() - 13.531µs
[TRACE] 11:13:25.779 shell.go:GOOS() - 110ns
[DEBUG] 11:13:25.779 shell.go:CommandPath:596 → /usr/bin/git
[TRACE] 11:13:25.779 shell.go:CommandPath(git) - 66.481µs
[TRACE] 11:13:25.779 shell.go:HasCommand(git) - 72.035µs
[DEBUG] 11:13:25.779 shell.go:Pwd:400 → /home/lompask
[TRACE] 11:13:25.779 shell.go:Pwd() - 7.861µs
[ERROR] 11:13:25.779 shell.go:HasParentFilePath:729 → stat .git: no such file or directory
[TRACE] 11:13:25.779 shell.go:HasParentFilePath(.git) - 53.381µs
[TRACE] 11:13:25.779 shell.go:Flags() - 148ns
[DEBUG] 11:13:25.779 shell.go:Pwd:400 → /home/lompask
[TRACE] 11:13:25.779 shell.go:Pwd() - 5.338µs
[TRACE] 11:13:25.779 shell.go:GOOS() - 117ns
[TRACE] 11:13:25.779 shell.go:ExecutionTime() - 139ns
[TRACE] 11:13:25.779 shell.go:Flags() - 68ns
[DEBUG] 11:13:25.779 shell.go:Pwd:400 → /home/lompask
[TRACE] 11:13:25.779 shell.go:Pwd() - 4.853µs
[TRACE] 11:13:25.779 shell.go:GOOS() - 119ns
[TRACE] 11:13:25.779 shell.go:ErrorCode() - 165ns
[TRACE] 11:13:25.779 shell.go:TemplateCache() - 104ns
[TRACE] 11:13:25.779 shell.go:Shell() - 189ns
[TRACE] 11:13:25.780 shell.go:TemplateCache() - 325ns
[DEBUG] 11:13:25.780 shell.go:Getenv:392 → NO DATA
[TRACE] 11:13:25.780 shell.go:Getenv(XDG_CACHE_HOME) - 7.035µs
[TRACE] 11:13:25.780 shell_unix.go:CachePath() - 26.338µs
[TRACE] 11:13:25.780 shell.go:Flags() - 149ns
```

--- a/src/shell/scripts/omp.fish
+++ b/src/shell/scripts/omp.fish
@@ -39,7 +39,8 @@ function fish_prompt
     set_poshcontext
     # validate if the user cleared the screen
     set --local omp_cleared false
-    if test (history | head -1) = "clear"
+    set --local last_command (history search --max 1)
+    if test "$last_command" = "clear"
       set omp_cleared true
     end
     ::OMP:: print primary --config $POSH_THEME --shell fish --error $omp_status_cache --execution-time $omp_duration --stack-count $omp_stack_count --shell-version $FISH_VERSION --cleared=$omp_cleared

